### DISCUSSION TITLE
how to change language in UI

### DISCUSSION THREAD

**Discussion by fanxting** [dcm4che/dcm4chee-arc-light#3259] [Q&A] [answered]
hi
new here.
I have already installed the dcm4chee-arc-light following the installation guides
https://github.com/dcm4che/dcm4chee-arc-light/wiki/Installation
But I can't see the options to choose language
![123ui](https://user-images.githubusercontent.com/15911131/122855526-f286ee80-d347-11eb-9d45-b0e96cd2e67e.jpg)

Did I do anything wrong?

**Comment by fanxting** [comment, marked as answer]
![Uploading ui.png…]()

**Reply by shral** [reply]
Hi, 
on default, there is no language configured, but you can do that like this:
Go to Configuration->Devices->Click on the mane archive device (normally called "dcm4chee-arc")->Open "Extensions"->Click on "Edit extension"->open "Child objects"-> on the "UI Configuration" click on the first child "default"->On the "Language Config" click on + and create a language config->Open the "Attributes" block-> Give some name to the Language Config Name->on Available Languages choose English and the language that you want to be available,  (Only 7 languages are currently available and only English is good tested one the others are on alpha version and not yet much tested)
Which language would you need anyhow?

Best Regards
Shefki Esadi

**Reply by fanxting** [reply]
Chinese， I would like to use Chinese simpled。
It seems to be unavailable

**Comment by fanxting** [comment]
What should I do if I want to add Chinese simpled to the UI config?
Thank you
Fan

**Reply by shral** [reply]
Chinese is not available currently, but if you want to contribute, you are more than welcome. The language files are located in the [dcm4chee-arc-lang](https://github.com/dcm4che/dcm4chee-arc-lang) project. So you may start by copieng the [en.json](https://github.com/dcm4che/dcm4chee-arc-lang/blob/master/src/main/webapp/assets/locale/en.json) file and generating a Chinese JSON file and starting to translate the texts. If you want that the **configurations are also translated** you have to translate the [schemas](https://github.com/dcm4che/dcm4chee-arc-lang/tree/master/src/main/webapp/assets/schema) too (by copping all the JSON schemas in a folder with the name as the language code). (Or for the beginning just download the en.json translate it and send us back and we will prepare the rest)

**Reply by fanxting** [reply]
[locale_ChineaseSimpled.zip](https://github.com/dcm4che/dcm4chee-arc-light/files/7258083/locale_ChineaseSimpled.zip)

**Comment by fanxting** [comment]
Hi
Please find the attached files
already did the translation

[locale_ChineaseSimpled.zip](https://github.com/dcm4che/dcm4chee-arc-light/files/7258058/locale_ChineaseSimpled.zip)

**Reply by shral** [reply]
Hi,
thank you for your contribution! I see that you translated the schemas as CSV-files, could you transform them to JSON-files as the others are?

Best Regards
Shefki Esadi

**Comment by gunterze** [comment]
I will take care to convert the provided CSV files to corresponding properties (s. https://github.com/dcm4che/dcm4chee-arc-lang/issues/57 ) and JSON schema files and push it to https://github.com/dcm4che/dcm4chee-arc-lang . You should now also have write access to https://github.com/dcm4che/dcm4chee-arc-lang.

**Reply by gunterze** [reply]
Had to adjust comma delimiters and quotes in several CSV files to be able to convert them to properties and JSON schema files.
Imported by commit https://github.com/dcm4che/dcm4chee-arc-lang/commit/84ea928c61ff3e683a63d36ea1dd0150360664df

Opened https://github.com/dcm4che/dcm4chee-arc-lang/issues/60 . Including `#60` in the message text of your commits to  https://github.com/dcm4che/dcm4chee-arc-lang to associate them to that issue.

Have you received an invitation to dcm4chee-arc-lang contributors? You will have to confirm it before you can commit updates and I can assign you  https://github.com/dcm4che/dcm4chee-arc-lang/issues/60 .

**Reply by fanxting** [reply]
I confirmed the invitation.
Thanks

**Reply by fanxting** [reply]
Hi there,

we already got the translation jsons. What should I do to put them into the system here:
![图片](https://user-images.githubusercontent.com/15911131/137826032-fbd005d4-6417-458c-a94d-d168aa09f803.png)

Dcm4chee running on docker.

What should I do now?

Thanks

**Comment by fanxting** [comment]
Hi there,

we already got the translation jsons. What should I do to put them into the system here:
![图片](https://user-images.githubusercontent.com/15911131/137826032-fbd005d4-6417-458c-a94d-d168aa09f803.png)

Dcm4chee running on docker.

What should I do now?

Thanks
